.github/workflows/ci.yml

```diff
@@ -0,0 +1,93 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  CI:
+    env:
+      LOCAL_CI_TESTS_CACHEDIR: ${{ GITHUB.WORKSPACE }}/cachedir
+      LOCAL_CI_TESTS_GITDIR: "${{ GITHUB.WORKSPACE }}/moodle"
+      LOCAL_CI_TESTS_PHPCS_DIR: "${{ GITHUB.WORKSPACE }}/moodle-local_codechecker/moodle"
+      LOCAL_CI_TESTS_DBLIBRARY: native
+      LOCAL_CI_TESTS_DBTYPE: mysqli
+      LOCAL_CI_TESTS_DBHOST: 127.0.0.1
+      LOCAL_CI_TESTS_DBUSER: moodle
```

Comment:
Hi @andrewnicols,
I wonder if that should be `root` since https://docs.travis-ci.com/user/database-setup/#mysql tells that their `travis` user is equivalent to `root` in their way to setup the MySQL service i.e. no need to create an unprivileged `moodle` user that will have all the powers but limited to the `example` DB - it will fail in creating a new DB (`ci_installed_6678_0`) - which is not the actual MySQL connection error (`ERROR 2003 (HY000): Can't connect to MySQL server on '127.0.0.1' (111)`).

HTH,
Matteo